sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -0,0 +1,593 @@
+    const response: HttpOperationResponse = await this._listResources(
+      topicPath + "/Subscriptions/",
+      listRequestOptions,
+      this.subscriptionResourceSerializer
+    );
+    return this.buildResponse(response);
+  }
+
+  /**
+   * Creates a rule.
+   * @param topicPath
+   * @param subscriptionPath
+   * @param rule
+   * @param ruleOptions
+   */
+  async createRule(
+    topicPath: string,
+    subscriptionPath: string,
+    rule: string,
+    ruleOptions: RuleOptions
+  ): Promise<RuleResult> {
+    const fullPath = this.getRulePath(topicPath, subscriptionPath, rule);
+
+    const response: HttpOperationResponse = await this._putResource(
+      fullPath,
+      ruleOptions,
+      false,
+      this.ruleResourceSerializer
+    );
+    return this.buildResponse(response);
+  }
+
+  /**
+   * Updates a rule.
+   * @param topicPath
+   * @param subscriptionPath
+   * @param rule
+   * @param ruleOptions
+   */
+  async updateRule(
+    topicPath: string,
+    subscriptionPath: string,
+    rule: string,
+    ruleOptions: RuleOptions
+  ): Promise<RuleResult> {
+    const fullPath = this.getRulePath(topicPath, subscriptionPath, rule);
+
+    const response: HttpOperationResponse = await this._putResource(
+      fullPath,
+      ruleOptions,
+      true,
+      this.ruleResourceSerializer
+    );
+    return this.buildResponse(response);
+  }
+
+  /**
+   * Deletes a rule.
+   * @param topicPath
+   * @param subscriptionPath
+   * @param rule
+   */
+  async deleteRule(topicPath: string, subscriptionPath: string, rule: string): Promise<RuleResult> {
+    const fullPath = this.getRulePath(topicPath, subscriptionPath, rule);
+    const response: HttpOperationResponse = await this._deleteResource(
+      fullPath,
+      this.ruleResourceSerializer
+    );
+    return this.buildResponse(response);
+  }
+
+  /**
+   * Gets a rule.
+   * @param topicPath
+   * @param subscriptionPath
+   * @param rule
+   */
+  async getRule(topicPath: string, subscriptionPath: string, rule: string): Promise<RuleResult> {
+    const fullPath = this.getRulePath(topicPath, subscriptionPath, rule);
+    const response: HttpOperationResponse = await this._getResource(
+      fullPath,
+      this.ruleResourceSerializer
+    );
+    return this.buildResponse(response);
+  }
+
+  /**
+   * Lists existing rules.
+   * @param topicPath
+   * @param subscriptionPath
+   * @param listRequestOptions
+   */
+  async listRules(
+    topicPath: string,
+    subscriptionPath: string,
+    listRequestOptions?: ListRequestOptions
+  ): Promise<ListRulesResult> {
+    const fullPath = this.getSubscriptionPath(topicPath, subscriptionPath) + "/Rules/";
+    const response: HttpOperationResponse = await this._listResources(
+      fullPath,
+      listRequestOptions,
+      this.ruleResourceSerializer
+    );
+    return this.buildResponse(response);
+  }
+
+  /**
+   * @param queuePath
+   */
+  formatDeadLetterPath(queuePath: string): string {
```

Comment:
Is this method meant to be public, and to return the dead letter queue path for the customer to use? I'm not sure when I would call this method, and `format` doesn't seem like the right prefix (maybe `get`) to get a dead letter queue path.